front/src/modules/trainschedule/components/ImportTrainSchedule/ImportTrainScheduleConfig.tsx

```diff
@@ -144,16 +147,221 @@ const ImportTrainScheduleConfig = ({
+          departureTime = cleanTimeFormat(departureTime);
+        }
+
+        if (!ocpRef) {
+          console.error('ocpRef is null or undefined');
+          return null;
+        }
+        const operationalPoint = cichDict[ocpRef];
+
+        if (!operationalPoint) {
+          return null; // Skip step if not found in the cichDict
+        }
+        //! We add 87 to the CI code to create the UIC. It is France specific and will break if used in other countries.
+        const uic = Number(`87${operationalPoint.ciCode}`); // Add 87 to the CI code to create the UIC
+        const { chCode } = operationalPoint;
+        const formattedArrivalTime = `${startDate} ${arrivalTime}`;
+        const formattedDepartureTime = `${startDate} ${departureTime}`;
+
+        return {
+          id: nextId(),
+          uic,
+          chCode,
+          name: ocpRef,
+          arrivalTime: cleanTimeFormat(formattedArrivalTime),
+          departureTime: cleanTimeFormat(formattedDepartureTime),
+        } as Step;
+      })
+      .filter((step): step is Step => step !== null);
+
+  const mapTrainNames = (trainSchedules: ImportedTrainSchedule[], trains: Element[]) => {
+    const trainPartToTrainMap: Record<string, string> = {};
+
+    trains.forEach((train) => {
+      const trainPartRef = train.getElementsByTagName('trainPartRef')[0]?.getAttribute('ref');
+      const trainName = train.getAttribute('name') || '';
+      if (trainPartRef) {
+        trainPartToTrainMap[trainPartRef] = trainName;
+      }
+    });
+
+    const updatedTrainSchedules = trainSchedules.map((schedule) => {
+      const mappedTrainNumber = trainPartToTrainMap[schedule.trainNumber] || schedule.trainNumber;
+
+      return {
+        ...schedule,
+        trainNumber: mappedTrainNumber,
+      };
+    });
+
+    return updatedTrainSchedules;
+  };
+
+  const parseRailML = async (xmlDoc: Document): Promise<ImportedTrainSchedule[]> => {
+    const trainSchedules: ImportedTrainSchedule[] = [];
+
+    // Initialize localCichDict
+    const localCichDict: Record<string, CichDictValue> = {};
+
+    // Get all the operational points from the XML file
+    const infrastructures = Array.from(xmlDoc.getElementsByTagName('infrastructure'));
+
+    infrastructures.forEach((infrastructure) => {
+      const operationControlPoints = Array.from(
+        infrastructure.getElementsByTagName('operationControlPoints')
+      ).concat(Array.from(infrastructure.getElementsByTagName('OperationControlPoints')));
+
+      operationControlPoints.forEach((operationControlPoint) => {
+        const ocps = Array.from(operationControlPoint.getElementsByTagName('ocp'));
+
+        ocps.forEach((ocp) => {
+          const id = ocp.getAttribute('id');
+          const code = ocp.getAttribute('code');
+
+          if (id && code) {
+            const { ciCode, chCode } = extractCiChCode(code);
+            localCichDict[id] = { ciCode, chCode };
+          }
+        });
+      });
+    });
```

Comment:
As we just use ocp is it possible to just access ocp without using nested loops ?